### ISSUE TITLE
When ES5 version is used babel-runtime is required

### ISSUE THREAD

**Issue by kirill-konshin** [wheresrhys/fetch-mock#272]
Per documentation https://babeljs.io/docs/plugins/transform-runtime/#technical-details you must include `babel-runtime` in `dependencies` of your library, otherwise it produces errors like this:

```
Error: Cannot find module 'babel-runtime/core-js/promise'
```

**Comment by wheresrhys** [comment]
Can you try v6.1.0-beta.2 please. This has been an issue for users for a few weeks, but I had problems with the build preventing a fresh release. Finally managed to get a new release out though

**Comment by kirill-konshin** [comment]
Contents of `client.js`:

```
require('core-js/modules/es6.typed.array-buffer');
require('core-js/modules/es6.typed.data-view');
require('core-js/modules/es6.typed.int8-array');
require('core-js/modules/es6.typed.uint8-array');
require('core-js/modules/es6.typed.uint8-clamped-array');
```

Contents of installed `package.json`:

```
  "dependencies": {
    "glob-to-regexp": "^0.3.0",
    "path-to-regexp": "^2.1.0"
  },
```

So it still fails.

**Comment by kirill-konshin** [comment]
This is not right, think about, compatibility, even per babel-runtime doc a lib should install such dependency by itself, as dependency, not as peer dependency. If needed it will be lifted up by NPM.

**Comment by wheresrhys** [comment]
On the other hand, having babel-polyfill present is an environment set up choice, very similar to letting the end user choose which version of node-fetch to install. I prefer to be as unopinionated as possible here, with the cost that use in older environments may take a little extra set up, but the benefit that fetch-mock is less likely to suffer from strange conflicts with other libraries. Probably at least 50% of the issues raised here are related to interoperability with various other dev tools - my gut feeling is that the less I bundle into fetch-mock, the easier that is to manage.

Thanks for sharing your thoughts though. It's good to be forced to think carefully about these things

**Comment by kirill-konshin** [comment]
I am not talking about babel-polyfill. Users should polyfill their own env, that's right. But here we are talking about babel-runtime, a DIRECT DEPENDENCY that is a result of code compilation. I must be a part of package dependencies, bc other packages may require different version. You can compile with different set of settings to avoid that.

**Comment by jimsimon** [comment]
@kirill-konshin is correct here.  If the library needs babel-runtime to run in production then it should have a dependency on it.  At the very least, it should be a peerDependency.

In my case, I have two different dependencies resolving to two semantically different versions of core-js.  Fetch-mock ultimately tries to use the version of core-js that ends up in the top-level of my node_modules, which fails my webpack build because it's core-js 1.x and fetch-mock needs 2.x.

**Comment by wheresrhys** [comment]
It is a dependency now. Yep, kirill was right

**Comment by jimsimon** [comment]
Hm...I'm still running into some issues, though I wonder if what I'm seeing is an npm bug maybe.
